### ISSUE TITLE
console running WSL is unresponsive when running program with lots of output

### ISSUE THREAD

**Issue by matthargett** [microsoft/terminal#198]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
Microsoft Windows [Version 10.0.17134.81]

```
matt@DESKTOP-JN6UUK7:~$ uname -a
Linux DESKTOP-JN6UUK7 4.4.0-17134-Microsoft #81-Microsoft Sun May 20 01:14:00 PST 2018 x86_64 x86_64 x86_64 GNU/Linux
matt@DESKTOP-JN6UUK7:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04 LTS
Release:        18.04
Codename:       bionic
```

This is on a Surface Book 2, happens when on battery or connected to power. There is no external display attached.

* What you're doing and what's happening: (Copy & paste specific commands and their output, or include screen shots)

In WSL for Ubuntu 18.04, do the following:

1. Install nvs using the Linux instructions and choose 8.11.x as the version to install: https://github.com/jasongin/nvs#mac-linux
2. `sudo apt-get install git` and clone react-native-windows. `git clone https://github.com/Microsoft/react-native-windows`
3. cd into the react-native-windows directory and run `npm install`
4. run `node_modules/.bin/eslint .`

* What's wrong / what should be happening instead:

eslint has thousands of warnings. Try and control+C to break out of eslint. It won't work.

try to click the close button on the WSL console Window. it doesn't response. the eslint output just keeps going and going.

walk away for 20-30 minutes. when eslint stops running, you'll see the control+C and/or window close finally take effect.

**Comment by matthargett** [comment]
Is there anything I can do to help get this resolved? Were there any difficulties reproducing the issue?

**Comment by kaddkaka** [comment]
This is a very annoying issue for me and time thief since it can take 10 seconds to switch in tmux while some process is printing a lot in the current window.

Have you looked into this issue?

**Comment by DHowett-MSFT** [comment]
@kaddkaka we're looking into perf all the time. What version of the traditional windows console host (because that's what this bug was reported against) are _you_ using?

**Comment by kaddkaka** [comment]
> @kaddkaka we're looking into perf all the time. What version of the traditional windows console host (because that's what this bug was reported against) are _you_ using?

Where can I see which version I have?

**Comment by zadjii-msft** [comment]
![image](https://user-images.githubusercontent.com/18356694/80605467-be8c9380-89f8-11ea-860c-d83a83cfb149.png)
![image](https://user-images.githubusercontent.com/18356694/80605482-c77d6500-89f8-11ea-84e3-7de5b94cc601.png)

**Comment by kaddkaka** [comment]
> EDIT: derp this is a conhost thread
> 
> to get the _conhost_ version, run `winver` from `cmd` or `powershell`

![image](https://user-images.githubusercontent.com/4508793/80606079-257b6e00-8a34-11ea-8f5d-f4838c08b97c.png)

**Comment by kaddkaka** [comment]
> EDIT: derp this is a conhost thread
What was that first screenshot from? Is it a different Windows terminal?

**Comment by zadjii-msft** [comment]
> What was that first screenshot from? Is it a different Windows terminal?

Uh, yea that screenshot was from the [Windows Terminal](https://www.microsoft.com/en-us/p/windows-terminal-preview/9n0dx20hk701), which is hosted on [this repo](https://github.com/microsoft/terminal)

**Comment by squrky** [comment]
Hi all, 

Has there been any progress towards a fix for this? The problem seems to be your common or garden starvation of user input when large amounts of output are being produced. I'm able to reproduce this issue with the following software versions: 

- Windows: 10.0.19042.1165
- Windows Terminal: 1.10.1933.0
- ConHost: 10.0.19041.964
- Ubuntu: Linux threadbear 5.10.16.3-microsoft-standard-WSL2 #1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

As the original poster suggested, this problem is obstructive for doing real work in the terminal, since it prevents job control precisely when it is needed the most. Let me know if I can offer any other information or help in any other way to move this forward. 

Thanks.

**Comment by miniksa** [comment]
@lhecker didn't you improve something around this recently with the ticket lock?

**Comment by lhecker** [comment]
@miniksa @squrky Oh yeah I fixed this in https://github.com/microsoft/terminal/pull/10653. I had my terminal freeze for up to a minute under certain conditions. The fix will be released very soon in Windows Terminal Preview v1.11.2421.0 (I wonder if it‘ll be released today… \*wink\*).

I‘m very confident in my fix so I‘ll take the liberty to close this issue.
